bugtool/cmd/root.go

```diff
@@ -315,6 +319,39 @@ func runAll(commands []string, cmdDir string, k8sPods []string) {
 			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown)
 			continue
 		}
+
+		if strings.Contains(cmd, "xfrm") {
+			var output []string
+			execResult, _ := execCommand(cmd)
+			// splitting the result as an array of string
+			lines := strings.Split(execResult, "\n")
+			// list of substrings to search for in lines
+			substrings := [...]string{"auth", "enc", "aead", "auth-trunc", "comp"}
+			for _, line := range lines {
+				for _, substr := range substrings {
+					// checking if the line contains our substring
+					matched, _ := regexp.MatchString(substr, line)
+					if matched {
+						// looking for hex substring which starts with '0x'
+						for i, splits := range strings.Fields(line) {
+							hexFound, _ := regexp.MatchString("0x", splits)
+							if hexFound {
+								splits := strings.Fields(line)
+								pass := strings.Fields(line)[i]
+								hash := md5.Sum([]byte(pass))
+								splits[i] = "[hash: " + hex.EncodeToString(hash[:]) + "]"
+								line = strings.Join(splits, " ")
+							}
+						}
+					}
```

Comment:
If the regex is not matched, this means we have a line with a pattern like `auth` but we were not able to parse the key. In that case, can we remove the line entirely from the final output? I'd rather miss the line (this is what we have already) than leak the keys accidentally.

Reply:
fixed this .